src/main/java/org/ow2/proactive/catalog/repository/BucketRepositoryImpl.java

```diff
@@ -51,28 +51,35 @@ public class BucketRepositoryImpl implements BucketRepositoryCustom {
     public List<Object[]> findBucketContainingKindListAndContentTypeAndObjectName(List<String> kindList,
             String contentType, String objectName) {
 
-        return em.createQuery(buildCriteriaQuery(null, kindList, contentType, objectName)).getResultList();
+        return em.createQuery(buildCriteriaQuery(null, kindList, contentType, objectName, null, null)).getResultList();
     }
 
     @Override
-    public List<Object[]> findBucketByOwnerContainingKindListAndContentTypeAndObjectName(List<String> owners,
-            List<String> kindList, String contentType, String objectName) {
-
-        return em.createQuery(buildCriteriaQuery(owners, kindList, contentType, objectName)).getResultList();
+    public List<Object[]> findBucketByOwnerContainingKindListAndContentTypeAndObjectNameAndLastCommittedTimeInterval(
```

Comment:
Maybe we should rename this method and avoid redundancy of filtering methods by refactoring them  but I don't know how hard it is to refactor this

- findBucketByOwnerContainingKindListAndContentTypeAndObjectName
- findBucketByOwnerContainingKindList
- findBucketByOwnerContainingKindListAndContentTypeAndObjectNameAndLastCommittedTimeInterval
- etc...

This would have been acceptable in a test code, but since, practically we have a single endpoint that uses multiple filters, its better to use one generic filterBuckets() method with a list of filters or a BucketFilters entity....
what do you think @fviale 

Because otherwise, this code looks very complex and a little confusing for future developers.
Especially that we don't use filters this way in other micro-services and in the scheduler itself

Reply:
It's true for methods in the repositories interfaces too like `BucketRepositoryCustom`